src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -189,10 +233,21 @@ def lower_to_language_level(self):
                         "found '{0}'.".format(type(arg).__name__))
 
         symbol_table = self.scope.symbol_table
-        routine_symbol = self.psylayer_routine_symbol
-        container_symbol = routine_symbol.interface.container_symbol
-        symbol_table.add(container_symbol)
-        symbol_table.add(routine_symbol)
+
+        container_tag = self.psylayer_container_root_name
+        try:
+            container_symbol = symbol_table.lookup_with_tag(container_tag)
+        except KeyError:
+            container_symbol = symbol_table.new_symbol(
+                root_name=container_tag, tag=container_tag,
+                symbol_type=ContainerSymbol)
+
+        routine_tag = self.psylayer_routine_root_name
+        interface = GlobalInterface(container_symbol)
+        routine_symbol = symbol_table.new_symbol(
+            root_name=routine_tag, tag=routine_tag, symbol_type=RoutineSymbol,
+            interface=interface)
+
```

Comment:
You are correct, but this was the case before this work as well (or at least we would produce incorrect code previously). The issue is that we are not fully there at the moment. The PSy-layer is still receiving information in the old way (a hierarchy called `invoke_info` that is passed into `GOPSy()`) but we have updated the way the algorithm layer is generated (using PSyIR). The step after this one will be to pass the algorithm psyir into the psy-layer instead of `invoke_info` (or probably as well as, in the first instance) and then the psy-layer will be able to pick up the generated names from the algorithm psyir, rather than using `invoke_info`. This is mentioned in #753 but probably not prominently enough.

Reply:
Ok, so can you add a TODO to #753 here saying that this can currently produce a routine_symbol that doesn't match the PSy-layer routine name.

Reply:
Added, and also put a comment in #753.